fe/fe-core/src/main/java/org/apache/doris/nereids/jobs/executor/Rewriter.java

```diff
@@ -151,10 +151,8 @@ public class Rewriter extends AbstractBatchJobExecutor {
                     // we need run the following 2 rules to make AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION work
                     bottomUp(new PullUpProjectUnderApply()),
                     topDown(new PushdownFilterThroughProject()),
-                    costBased(
-                            custom(RuleType.AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION,
-                                    AggScalarSubQueryToWindowFunction::new)
-                    ),
+                    custom(RuleType.AGG_SCALAR_SUBQUERY_TO_WINDOW_FUNCTION,
```

Comment:
it is not always to have positive return for this costbased rule, under our current incomplete code model, we also need a mechanism to control this for on or off.

Reply:
good comments.
we will make framework more agile to support change rules at runtime.